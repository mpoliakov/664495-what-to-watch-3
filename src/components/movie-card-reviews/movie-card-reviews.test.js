import React from 'react';
import renderer from 'react-test-renderer';
import MovieCardReviews from './movie-card-reviews.jsx';

it(`<MovieCardReviews/> is rendered correctly`, () => {
  const reviews = [
    {
      id: 1,
      user: {
        id: 12,
        name: `Isaac`
      },
      rating: 2.6,
      comment: `Poised to be an instant classic, almost everything about this film is phenomenal - the acting, the cinematography, the discography, etc.`,
      date: `2020-03-06T05:40:57.360Z`
    },
    {
      id: 2,
      user: {
        id: 16,
        name: `Mollie`
      },
      rating: 5.9,
      comment: `This movie is just plain bad. There must be some big payola going round this awards season. Badly written, average acting at best, all the characters are unrelatable and inlikeable. 2 hours of my life wasted.`,
      date: `2020-03-17T05:40:57.360Z`
    }
  ];

  const component = renderer.create(<MovieCardReviews reviews={reviews} />).toJSON();
  expect(component).toMatchSnapshot();
});
