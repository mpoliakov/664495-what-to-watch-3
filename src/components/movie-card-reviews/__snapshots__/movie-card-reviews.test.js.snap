// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieCardReviews/> is rendered correctly 1`] = `
<div
  className="movie-card__reviews movie-card__row"
>
  <div
    className="movie-card__reviews-col"
  >
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          Poised to be an instant classic, almost everything about this film is phenomenal - the acting, the cinematography, the discography, etc.
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            Isaac
          </cite>
          <time
            className="review__date"
            dateTime="2020-03-06T05:40:57.360Z"
          >
            March 6, 2020
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        2.6
      </div>
    </div>
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          This movie is just plain bad. There must be some big payola going round this awards season. Badly written, average acting at best, all the characters are unrelatable and inlikeable. 2 hours of my life wasted.
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            Mollie
          </cite>
          <time
            className="review__date"
            dateTime="2020-03-17T05:40:57.360Z"
          >
            March 17, 2020
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        5.9
      </div>
    </div>
  </div>
</div>
`;
